app/controllers/api/v1/venues_controller.rb

```diff
@@ -0,0 +1,53 @@
+# :reek:InstanceVariableAssumption :reek:NilCheck
+class Api::V1::VenuesController < ApplicationController
+  before_action :authenticate_user!, only: %i[index show]
+  before_action :organizer?, only: %i[create update destroy]
+  before_action :set_venue, only: %i[show update destroy]
+
+  def index
+    @venues = Venue.all
+    render json: @venues
+  end
+
+  def show
+    render json: @venue
+  end
+
+  def create
+    @venue = Venue.new(venue_params)
+    if @venue.save
+      render json: @venue, status: :created
+    else
+      render json: @venue.errors.full_messages, status: :unprocessable_entity
+    end
+  end
+
+  def update
+    if @venue.update_attributes(venue_params)
+      render json: @venue
+    else
+      render json: @venue.errors.full_messages, status: :unprocessable_entity
+    end
+  end
+
+  def destroy
+    @venue.destroy
+    head :no_content
+  end
+
+  private
+
+  def set_venue
+    @venue = Venue.find_by(id: params[:id])
+    head :not_found if @venue.nil?
+  end
+
+  def organizer?
+    authenticate_user!
```

Comment:
authenticate for all actions and you can avoid using it here.

Reply:
done